javascript/integration-tests/AddOffer.spec.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Gluwa, Inc. & contributors
+// SPDX-License-Identifier: The Unlicense
+
+import type { Balance } from '@polkadot/types/interfaces';
+
+import { Guid } from 'js-guid';
+
+import { ApiPromise, Keyring, WsProvider } from '@polkadot/api';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { PalletCreditcoinAskOrderId, PalletCreditcoinBidOrderId, PalletCreditcoinLoanTerms } from '@polkadot/types/lookup';
+
+import { POINT_01_CTC } from '../src/constants';
+
+import { randomEthAddress } from '../src/utils';
+import * as testUtils from './test-utils';
+
+
+describe('AddOffer', (): void => {
+  let api: ApiPromise;
+  let borrower, lender: KeyringPair;
+  let loanTerms: PalletCreditcoinLoanTerms;
+  let askGuid, bidGuid: string;
+  let askOrderId: PalletCreditcoinAskOrderId;
+  let bidOrderId: PalletCreditcoinBidOrderId;
+
+  const blockchain = 'Ethereum';
+  const expirationBlock = 100;
+
+  beforeEach(async () => {
+    process.env.NODE_ENV = 'test';
+
+    const provider = new WsProvider('ws://127.0.0.1:9944');
+
+    api = await ApiPromise.create({ provider });
+
+    const keyring = new Keyring({ type: 'sr25519' });
+
+    lender = keyring.addFromUri('//Alice', { name: 'Alice' });
+    const lenderAddress = randomEthAddress();
+    const lenderRegAddr = await testUtils.registerAddress(api, lenderAddress, blockchain, lender);
+    askGuid = Guid.newGuid().toString();
+
+    borrower = keyring.addFromUri('//Bob', { name: 'Bob' });
+    const borrowerAddress = randomEthAddress();
+    const borrowerRegAddr = await testUtils.registerAddress(api, borrowerAddress, blockchain, borrower);
+    bidGuid = Guid.newGuid().toString();
+
+    loanTerms = api.createType<PalletCreditcoinLoanTerms>('PalletCreditcoinLoanTerms', {
+      amount: 1_000,
+      interestRate: 100,
+      maturity: 100
+    });
+
+    askOrderId = await testUtils.addAskOrder(api, lenderRegAddr.addressId, loanTerms, expirationBlock, askGuid, lender);
+    bidOrderId = await testUtils.addBidOrder(api, borrowerRegAddr.addressId, loanTerms, expirationBlock, bidGuid, borrower);
+
+  }, 210000);
+
+  afterEach(async () => {
+    await api.disconnect();
+  });
+
+  it('fee is min 0.01 CTC', async (): Promise<void> => {
+    return new Promise((resolve, reject): void => {
+      const unsubscribe = api.tx.creditcoin
+        .addOffer(askOrderId, bidOrderId, expirationBlock)
```

Comment:
From what I can tell the IDs are:

```
askOrderId -> [100,"0xf157edfd22a8764f08648f8c01141b17709ba165371830b374d9d8114d6e63f4"]
bidOrderId -> [100,"0x718fb02e6a9a19d0a77fb1a4ce2df8abcdac11142bbbb4cc629ae909dd3a9928"]
```

and I get an error `creditcoin.NonExistentAskOrder: The specified ask order does not exist.`

@jfrank-summit, @nathanwhit can you inspect my code and tell me if I am passing the correct values here ? I have the feeling I'm not.

Reply:
I believe this is because you have expirationBlock set to 100. When i set to 100_000 it worked. Also, there is a bug where it lets you create expired Ask and Bid orders. I have created a ticket for it.

Reply:
Just to make this more clear: `expirationBlock` is the number of block after which the AskOrder will expire, right ? Originally I had a value of 5 here, which may be too small but a 100 blocks being mined in a few seconds between AddAskOrder and AddOffer doesn't sound very likely to me.

Reply:
It seems to take a long time to run, so at 3-6s blocktimes it does seem possible. It also needs to get through any other extrinsics such as registerAddress before the expiration block.  Have you tried increasing it?

Reply:
I've tried increasing expirationBlock to 10000. However locally I see

```
2022-04-07 11:49:18 ✅ Successfully mined block on top of: 0x8933…0ee4    
2022-04-07 11:49:18 ✨ Imported #7 (0xe36b…0304)    
2022-04-07 11:49:18 🙌 Starting consensus session on top of parent 0xe36b30585729d82d698d3bc5c34c4164fbb2d4ea7a8b06dc01d5de1252590304    
2022-04-07 11:49:18 🎁 Prepared block for proposing at 8 (16 ms) [hash: 0x33e89c80c1c5aab5ccdfe5ed1a82cea175cb6b3bf7180ede1c7db92ba2b1487f; parent_hash: 0xe36b…0304; extrinsics (2): [0x0bc7…1db3, 0xad33…2f5d]]    
2022-04-07 11:49:21 💤 Idle (0 peers), best: #7 (0xe36b…0304), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:49:22 ✨ Imported #8 (0x526e…37c5)    
2022-04-07 11:49:22 ✅ Successfully mined block on top of: 0xe36b…0304    
2022-04-07 11:49:22 🙌 Starting consensus session on top of parent 0x526e0582cbdd24feb03c62ba4c0b8bba66c9a587f8ef0fc99dcb856b03e937c5    
2022-04-07 11:49:22 🎁 Prepared block for proposing at 9 (10 ms) [hash: 0xf158e4d9b8aac91eb90f92b8a75fc954cb16effcd96d34b938acf958d14a217a; parent_hash: 0x526e…37c5; extrinsics (1): [0xec9a…1d53]]    
2022-04-07 11:49:26 💤 Idle (0 peers), best: #8 (0x526e…37c5), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:49:31 💤 Idle (0 peers), best: #8 (0x526e…37c5), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:49:36 💤 Idle (0 peers), best: #8 (0x526e…37c5), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:49:39 ✨ Imported #9 (0x620b…3e3c)    
2022-04-07 11:49:39 🙌 Starting consensus session on top of parent 0x620b412d73be9f2e4373980792f202a482ea52be9b0bc2cc4f8dbae934d43e3c    
2022-04-07 11:49:39 ✅ Successfully mined block on top of: 0x526e…37c5    
2022-04-07 11:49:39 🎁 Prepared block for proposing at 10 (14 ms) [hash: 0x055468f302466d1144db4cecd3206fec1286f4edae1bd4b505ce8757b26bcaaf; parent_hash: 0x620b…3e3c; extrinsics (2): [0xc41d…a94f, 0xda24…fb16]]    
2022-04-07 11:49:41 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:49:46 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:49:51 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:49:56 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:01 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:06 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:11 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:16 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:21 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:26 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:31 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
2022-04-07 11:50:36 💤 Idle (0 peers), best: #9 (0x620b…3e3c), finalized #0 (0x87fb…cb7a), ⬇ 0 ⬆ 0    
```

If I am reading this correctly then the node mined 9 blocks in total during the duration of this test. Nowhere near the 100 expiration boundary. 

I got a timeout failure locally so adjusting the timeout values now. 

Pushed the latest changes to see what happens in CI.

Reply:
looks like it worked